include/quadrilateral_gimp_element.tcc

```diff
@@ -0,0 +1,155 @@
+// Return natural nodal coordinates
+template <>
+inline Eigen::MatrixXd
+    mpm::QuadrilateralGIMPElement<2, 16>::natural_nodal_coordinates() const {
+  //! Natural coordinates of nodes
+  Eigen::Matrix<double, 16, 2> local_nodes;
+  // clang-format off
+  local_nodes << -1., -1.,
+                  1., -1.,
+                  1.,  1.,
+                 -1.,  1.,
+                 -3., -3.,
+                 -1., -3.,
+                  1., -3.,
+                  3., -3.,
+                  3., -1.,
+                  3.,  1.,
+                  3.,  3.,
+                  1.,  3.,
+                 -1.,  3.,
+                 -3.,  3.,
+                 -3.,  1.,
+                 -3., -1.;
+  // clang-format on
+  return local_nodes;
+}
+
+//! Return shape functions of a 16-node Quadrilateral GIMP Element at a given
+//! local coordinate
+template <>
+inline Eigen::VectorXd mpm::QuadrilateralGIMPElement<2, 16>::shapefn(
+    const Eigen::Matrix<double, 2, 1>& xi, const VectorDim& particle_size,
+    const Eigen::Matrix<double, 2, 1>& deformation_gradient) const {
+
+  //! Dimension
+  const unsigned Dim = 2;
+  //! Nodes in GIMP function
+  const unsigned Nfunctions = 16;
+
+  //! length of element in local coordinate
+  const double element_length = 2.;
+
+  //! Natural nodal coordinates
+  const Eigen::Matrix<double, 16, 2> local_nodes =
+      this->natural_nodal_coordinates();
+  //! To store shape functions
+  Eigen::Matrix<double, 16, 1> shapefn;
+  //! loop to iterate over nodes
+  for (unsigned n = 0; n < Nfunctions; ++n) {
+    //! local shape function in current plane (x, y or z)
+    Eigen::Matrix<double, 2, 1> sni;
+    //! loop to iterate over dimensions
+    for (unsigned i = 0; i < Dim; ++i) {
+      double ni = local_nodes(n, i);
+      double npni = xi(i) - ni;  // local particle  - local node
+
+      //! Conditional shape function statement see: Bardenhagen 2004
+      if (npni <= -element_length - particle_size(i)) {
```

Comment:
Please use `(` and `)` to group things together for example `if (npni <= -(element_length + particle_size(i)))`. Please do so for rest of the loop.